hal/tests/TESTS/pin_names/pinvalidate.py

```diff
@@ -14,29 +14,62 @@
 limitations under the License.
 """
 
-import os
-import re
 import argparse
-import pathlib
 import json
+import os
+import pathlib
+import re
+import sys
 from tabulate import tabulate
+from enum import Enum
+
+
+class ReturnCode(Enum):
+    """Return codes."""
+
+    SUCCESS = 0
+    ERROR = 1
+    INVALID_OPTIONS = 2
+
+
+class TestCaseError(Exception):
+    """An exception for test case failure."""
+
+
+class ArgumentParserWithDefaultHelp(argparse.ArgumentParser):
+    """Subclass that always shows the help message on invalid arguments."""
+
+    def error(self, message):
+        """Error handler."""
+        sys.stderr.write("error: {}\n".format(message))
+        self.print_help()
+        raise SystemExit(ReturnCode.INVALID_OPTIONS.value)
+
+
+def find_target(target_name=""):
+    """Find a target."""
+    mbed_os_root = os.path.abspath(
+        os.path.join(
+            pathlib.Path(__file__).parent.absolute(), "..", "..", "..", ".."
+        )
+    )
 
-def find_target(target_name = ''):
-    mbed_os_root = os.path.abspath(os.path.join(pathlib.Path(__file__).parent.absolute(), '..', '..', '..', '..'))
     targets = dict()
-    
-    try:
-        targets_json_file = open(os.path.join(mbed_os_root, 'targets', 'targets.json'))
-        target_data = json.load(targets_json_file)
-    except:
-        exit("Failed to open targets.json")
 
-    for root, dirs, files in os.walk(mbed_os_root):
+    targets_json_file = open(
+        os.path.join(mbed_os_root, "targets", "targets.json")
+    )
+
+    target_data = json.load(targets_json_file)
```

Comment:
I believe this could be simply:

```python
target_data = json.load((mbed_os_root / "targets" / "targets.json).open())
```
I do believe that the file is no longer needed once `json.load` does its thing. I think you could also do this instead:

```python
with (mbed_os_root / "targets" / "targets.json").open() as targets_json_file:
    target_data = json.load(targets_json_file)
```

This would take care of closing it once you're finished with it (which does already happen at the end of the script, but still).